packages/common/src/store/pages/chat/slice.ts

```diff
@@ -203,15 +238,21 @@ const slice = createSlice({
         return
       }
 
-      // Update the summary to include the min of next_cursor/next_count and
-      // prev_cursor/prev_count.
+      // Update the summary to include the max of next_cursor and
+      // min of prev_cursor.
       const existingSummary = state.chats.entities[chatId]?.messagesSummary
       const summaryToUse = { ...summary, ...existingSummary }
-      if (summary.next_count < (existingSummary?.next_count ?? Infinity)) {
+      if (
```

Comment:
is the reason `count` no longer works here because the `existingSummary.next_count` might be out of date? And would that cause us to recalculateHasTail on the wrong index on L274?

Reply:
I think we're probably good just from comparing before/after logic. Way to test would be to see if tails randomly pop in/out when websocket reconnects and re-sends latest messages.

Reply:
When you first load the chat, you load from the most recent, so `next_count` is zero. When you refresh, you also load from the most recent, _but the most recent has changed_. The `next_count` is still 0, but the cursor changed. If we just looked at `next_count` then we won't update the cursor, and we'll keep refetching anything newer than that cursor over and over.

Reply:
tl;Dr `next_count` is always zero, but the cursor changes